src/components/header/Header.jsx

```diff
@@ -1,3 +1,78 @@
+import { FaBars, FaTimes } from "react-icons/fa";
+
+import { SearchBox } from "./SearchBox";
+import { NavButtons } from "./NavButtons";
+
+import "./header.css";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import { useFixBody } from "../../hooks";
+
+import { BtnIcon } from "../buttons";
+
 export function Header() {
-  return <>Header</>;
+  const [isSideMenuOpen, setIsSideMenuOpen] = useState(false);
+
+  function toggleSideMenu() {
+    console.log("toggling side menu");
+    setIsSideMenuOpen((s) => !s);
+  }
+  useFixBody(isSideMenuOpen);
+
+  return (
+    <header>
+      <div className="hy-header-wrapper">
+        <nav className="hy-navbar px-sm pb-sm pt-1">
+          <div className="hy-navbar-links-container  mb-sm">
+            <div className="align-center">
+              <BtnIcon cnames="btn hy-hamburger-btn" onClick={toggleSideMenu}>
+                <FaBars />
+              </BtnIcon>
+              <Link to="/">
+                <div className="hy-logo">
+                  <span>
+                    Auto<span style={{ color: "#FFDE59" }}>Decore</span>
+                  </span>
+                </div>
+              </Link>
+            </div>
+            <NavButtons />
+          </div>
+          <SearchBox />
+        </nav>
+      </div>
+
+      <div className={`side-menu ${isSideMenuOpen ? "isOpen" : null}`}>
+        <div>
+          <img
+            src="https://static-assets-web.flixcart.com/www/linchpin/fk-cp-zion/img/profile-pic-male_4811a1.svg"
+            alt=""
```

Comment:
Good to have -> name to the alt attribute.

Reply:
The image is used just for decoration purposes.